collector/pkg/component/analyzer/cpuanalyzer/delete_tid.go

```diff
@@ -0,0 +1,74 @@
+package cpuanalyzer
+
+import (
+	"sync"
+	"time"
+)
+
+type tidDeleteQueue struct {
+	queueMutex sync.Mutex
+	queue      []deleteTid
+}
+
+type deleteTid struct {
+	pid      uint32
+	tid      uint32
+	exitTime time.Time
+}
+
+func (dq *tidDeleteQueue) GetFront() *deleteTid {
+	if len(dq.queue) > 0 {
+		return &dq.queue[0]
+	}
+	return nil
+}
+
+func (dq *tidDeleteQueue) Push(elem deleteTid) {
+	dq.queue = append(dq.queue, elem)
+}
+
+func (dq *tidDeleteQueue) Pop() {
+	if len(dq.queue) > 0 {
+		dq.queue = dq.queue[1:len(dq.queue)]
+	}
+}
+
+func (ca *CpuAnalyzer) InitTidDeleteQueue() {
+	ca.tidExpiredQueue = &tidDeleteQueue{queue: make([]deleteTid, 0)}
+}
```

Comment:
In Go, we usually "new" things in a single method.
```suggestion
func newTidDeleteQueue() *tidDeleteQueue {
	return &tidDeleteQueue{queue: make([]deleteTid, 0)}
}

ca.tidExpiredQueue = newTidDeleteQueue()
```